llvm/lib/Analysis/MemoryProfileInfo.cpp

```diff
@@ -347,3 +347,24 @@ template <> uint64_t CallStack<MDNode, MDNode::op_iterator>::back() const {
   return mdconst::dyn_extract<ConstantInt>(N->operands().back())
       ->getZExtValue();
 }
+
+MDNode *MDNode::getMergedMemProfMetadata(MDNode *A, MDNode *B) {
+  if (!(A && B)) {
+    return A ? A : B;
+  }
+
+  // TODO: Support more sophisticated merging, such as selecting the one with
+  // more bytes allocated, or implement support for carrying multiple allocation
+  // leaf contexts. For now, keep the first one.
+  return A;
```

Comment:
This logic is a little hard to read. Can this be replaced with --
```
if(A) return A;
return B;
```

Reply:
done